TypewiseAlert/TypewiseAlert.cs

```diff
@@ -1,90 +1,31 @@
 ﻿using System;
+using TypewiseAlert.ConcreteClass;
+using TypewiseAlert.Interface;
+using static TypewiseAlert.AlterterType;
 
 namespace TypewiseAlert
 {
-  public class TypewiseAlert
-  {
-    public enum BreachType {
-      NORMAL,
-      TOO_LOW,
-      TOO_HIGH
-    };
-    public static BreachType inferBreach(double value, double lowerLimit, double upperLimit) {
-      if(value < lowerLimit) {
-        return BreachType.TOO_LOW;
-      }
-      if(value > upperLimit) {
-        return BreachType.TOO_HIGH;
-      }
-      return BreachType.NORMAL;
+    public class TypewiseAlert : ITypewiseAlert
+    {
+        public BreachType breachType = new BreachType();
+        public bool IsCheckActiviated = false;
+        public bool IsAlertActivated = false;
+        public void checkAndAlert(AlertTarget alertTarget, BatteryCharacter batteryChar, double temperatureInC, ITriggerProcessor _triggerProcessor = null)
+        {
+            IBreach breach = new BreachClassifier();
+             breachType = breach.classifyTemperatureBreach(batteryChar.coolingType, temperatureInC);
+            
+            switch (alertTarget)
```

Comment:
This switch is not necessary, as the _triggerprocessor is already an interface.
The object, whether it is email or contrtoller is already determined out of scope of this method, hence switch is redundant!